server/sonar-server/src/main/java/org/sonar/server/test/index/TestIndexDefinition.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.test.index;
+
+import org.sonar.api.config.Settings;
+import org.sonar.server.es.EsUtils;
+import org.sonar.server.es.IndexDefinition;
+import org.sonar.server.es.NewIndex;
+
+public class TestIndexDefinition implements IndexDefinition {
+  public static final String INDEX = "tests";
+  public static final String TYPE = "test";
+  public static final String FIELD_UUID = "uuid";
+  public static final String FIELD_NAME = "name";
+  public static final String FIELD_STATUS = "status";
+  public static final String FIELD_MESSAGE = "message";
+  public static final String FIELD_STACKTRACE = "stacktrace";
+  public static final String FIELD_TYPE = "type";
+  public static final String FIELD_COVERAGE_BLOCKS = "coverageBlocks";
+  public static final String FIELD_COVERAGE_BLOCK_UUID = "uuid";
+  public static final String FIELD_COVERAGE_BLOCK_KEY = "key";
+  public static final String FIELD_COVERAGE_BLOCK_LINES = "lines";
+  public static final String FIELD_COVERAGE_BLOCK_COVERED_LINES = "coveredLines";
+
+  private final Settings settings;
+
+  public TestIndexDefinition(Settings settings) {
+    this.settings = settings;
+  }
+
+  @Override
+  public void define(IndexDefinitionContext context) {
+    NewIndex index = context.create(INDEX);
+
+    EsUtils.refreshHandledByIndexer(index);
+    EsUtils.setShards(index, settings);
+
+    NewIndex.NewIndexType nestedMapping = index.createType(TYPE);
+    nestedMapping.stringFieldBuilder(FIELD_COVERAGE_BLOCK_KEY).build();
```

Comment:
I thought we'd go with a denomalized schema. It means I'll have to request component to get the key then ?